### PR TITLE
Incremental patch for mmperf cuda

### PR DESCRIPTION
**Pull request by chudur-budur**
This is the first patch. This mainly does initial tests about running on cuda.

### CODE REVIEW

File: benchmark_sizes/benchmark_all_sizes.txt [hunk 1]
```diff
@@ -10,10 +10,10 @@
 384x256x256
 480x512x16
 480x512x256
-1024x1024x1024
+#1024x1024x1024
```
Comment (powderluv) [inline benchmark_sizes/benchmark_all_sizes.txt:13]: why ?

File: README.md [hunk 2]
```diff
@@ -51,6 +57,20 @@ To plot the results, you will need to install matplotlib.
 pip install matplotlib
 ```
 
+If you want to compile the code with `llvm`, you might need these:
```
Comment (powderluv) [inline README.md:60]: wouldn't just installing libclang-11-dev clang-11 liblld-11-dev be enough to pull in the deps?

File: make.sh [hunk 3]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
```
Comment (powderluv) [inline make.sh:1]: dont need this make.sh. Just add notes in the Readme.

File: matmul/CMakeLists.txt [hunk 4]
```diff
@@ -248,6 +255,10 @@ foreach(BACKEND ${BACKENDS})
       target_compile_definitions(${MATMUL} PRIVATE CUBLAS)
       target_include_directories(${MATMUL} PRIVATE ${CUDA_INCLUDE_DIRS})
       target_link_libraries(${MATMUL} PRIVATE ${CUDA_CUBLAS_LIBRARIES} ${CUDA_LIBRARIES})
+    elseif(${BACKEND} STREQUAL "cuda")
+      target_compile_definitions(${MATMUL} PRIVATE CUBLAS)
```
Comment (powderluv) [inline matmul/CMakeLists.txt:259]: should this be cuda or cublas ?

File: matmul/main.cc [hunk 5]
```diff
@@ -22,6 +22,10 @@
 #include "cuda_runtime.h"
 #include "cublas_v2.h"
 #include <sstream>
+#elif defined(CUDA)
+#include "cuda_runtime.h"
+#include "cublas_v2.h"
```
Comment (powderluv) [inline matmul/main.cc:27]: same as cublas ?

File: matmul/main.cc [hunk 6]
```diff
@@ -330,6 +366,14 @@ int main(int argc, char **argv) {
     CHECK_CUBLAS(cublasSgemm(handle, CUBLAS_OP_N, CUBLAS_OP_N, NDIM, MDIM, KDIM,
 			     &alpha, BB, LDB, AA, LDA, &beta, CC, LDC));
 #endif
+#elif defined(CUDA)
+#if defined(COLUMN_MAJOR)
+    CHECK_CUBLAS(cublasSgemm(handle, CUBLAS_OP_N, CUBLAS_OP_N, MDIM, NDIM, KDIM,
+			     &alpha, AA, LDA, BB, LDB, &beta, CC, LDC));
+#else
+    CHECK_CUBLAS(cublasSgemm(handle, CUBLAS_OP_N, CUBLAS_OP_N, NDIM, MDIM, KDIM,
+			     &alpha, BB, LDB, AA, LDA, &beta, CC, LDC));
```
Comment (powderluv) [inline matmul/main.cc:375]: I am confused why do we have cuda and cublas ?

File: benchmark_sizes/benchmark_all_sizes.txt (hunk 1 above)
Comment (chudur-budur) [reply benchmark_sizes/benchmark_all_sizes.txt:13]: Hi, I have made another new PR. Please disregard this PR. Thanks.

**Pull request closed without merging**